### PR TITLE
Freon Cold Fire

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Adds the visual effect of fire to freon burn reaction.
q

## Why It's Good For The Game
Cold fires are cool

## Changelog

:cl:
expansion: freon burn reaction now produces coldspots and cold cyan fires
/:cl:

### CODE REVIEW

**Comment by Ghilker** [comment]
Fire pretty
image

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 1]
```diff
@@ -401,10 +404,17 @@
 		new /obj/item/stack/sheet/hot_ice(holder)
 
 	SET_REACTION_RESULTS(freon_burn_rate * (1 + oxygen_burn_ratio))
+	air.reaction_results["fire"] += freon_burn_rate * (1 + oxygen_burn_ratio)
```
Comment (vincentiusvin) [inline code/modules/atmospherics/gasmixtures/reactions.dm:407]: We use reaction_results for the actual result of the reaction now. I suggest having this be set by SET_REACTION_RESULTS and then dealing with the effects in hotspot code.

Comment (vincentiusvin) [reply code/modules/atmospherics/gasmixtures/reactions.dm:407]: crap, the cleanup reintroduced the fires. one moment.

**Comment by LemonInTheDark** [comment]
Haven't checked as closely as I want to, but this looks a lot more sane then I expected. I'm glad there's no possible collision, thank you for that.

The purpose is primarily to serve as a visual representation right? or does it also boost the fire

**Comment by Ghilker** [comment]
> Haven't checked as closely as I want to, but this looks a lot more sane then I expected. I'm glad there's no possible collision, thank you for that.

Yes i tested it and there shouldn't be any issues with other kind of fires

> The purpose is primarily to serve as a visual representation right? or does it also boost the fire

is just visual representation, since it's bypassing the additional react() call is ignored

**Comment by Ghilker** [comment]
@LemonInTheDark duuuuuuude

**Comment by LemonInTheDark** [comment]
I read it properly. It's not doing anything weird, it's just also not really doing much with hotspot logic.
I still am of the opinion that stuff like this would be best handled with a component on the turf, in a similar way to water vapor. 

It's a bit more messy, but at least it'd be segmented. I know you're not so keen on that though, least I think. Is there a reason for that?

**Comment by Ghilker** [comment]
> I read it properly. It's not doing anything weird, it's just also not really doing much with hotspot logic. I still am of the opinion that stuff like this would be best handled with a component on the turf, in a similar way to water vapor.
> 
> It's a bit more messy, but at least it'd be segmented. I know you're not so keen on that though, least I think. Is there a reason for that?

i didn't do it because since it's a fire it should stay with fire logic, it uses some of the important part of hotspots so that it spreads properly, but i'll do other gas reactions in a similar way of water vapor since is closer to that kind of interaction.

**Comment by Ghilker** [comment]
@LemonInTheDark STolLU9B

File: code/modules/atmospherics/environmental/LINDA_fire.dm [hunk 2]
```diff
@@ -144,6 +151,8 @@
 		temperature = reference.temperature
 
 	// Handles the burning of atoms.
+	if(temperature < FIRE_MINIMUM_TEMPERATURE_TO_EXIST)
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_fire.dm:154]: Rather then sticking checks like this and `if(temperature < FREON_MAXIMUM_BURN_TEMPERATURE)` all over the place, it'd be better if you had a var on the hotspot called like "cold_burn" or whatever, and then set that on process. 
Would make the semi messy checks a bit esier to read

File: code/modules/atmospherics/environmental/LINDA_fire.dm [hunk 3]
```diff
@@ -231,24 +247,31 @@
...
+	if(!location.air || (INSUFFICIENT(/datum/gas/plasma) && INSUFFICIENT(/datum/gas/tritium) && INSUFFICIENT(/datum/gas/hydrogen) && INSUFFICIENT(/datum/gas/freon)) || INSUFFICIENT(/datum/gas/oxygen))
 		qdel(src)
 		return
 
 	perform_exposure()
 
 	if(bypassing)
 		icon_state = "3"
-		location.burn_tile()
+		if(location.air.temperature > FIRE_MINIMUM_TEMPERATURE_TO_EXIST)
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_fire.dm:267]:
```suggestion
		if(!cold_fire)
```
?

File: code/modules/atmospherics/environmental/LINDA_fire.dm [hunk 4]
```diff
@@ -274,7 +297,7 @@
 
 /obj/effect/hotspot/proc/on_entered(datum/source, atom/movable/arrived, atom/old_loc, list/atom/old_locs)
 	SIGNAL_HANDLER
-	if(isliving(arrived))
+	if(isliving(arrived) && temperature > FIRE_MINIMUM_TEMPERATURE_TO_EXIST)
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_fire.dm:300]: ditto

File: code/modules/atmospherics/environmental/LINDA_fire.dm [hunk 5]
```diff
@@ -231,24 +247,31 @@
...
-		location.burn_tile()
+		if(location.air.temperature > FIRE_MINIMUM_TEMPERATURE_TO_EXIST)
+			location.burn_tile()
 
 		//Possible spread due to radiated heat.
-		if(location.air.temperature > FIRE_MINIMUM_TEMPERATURE_TO_SPREAD)
+		if(location.air.temperature > FIRE_MINIMUM_TEMPERATURE_TO_SPREAD || location.air.temperature < FREON_MAXIMUM_BURN_TEMPERATURE)
 			var/radiated_temperature = location.air.temperature*FIRE_SPREAD_RADIOSITY_SCALE
+			if(location.air.temperature < FREON_MAXIMUM_BURN_TEMPERATURE)
+				radiated_temperature = location.air.temperature * COLD_FIRE_SPREAD_RADIOSITY_SCALE
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_fire.dm:274]:
```suggestion
		if(location.air.temperature > FIRE_MINIMUM_TEMPERATURE_TO_SPREAD || cold_fire)
			var/radiated_temperature = location.air.temperature*FIRE_SPREAD_RADIOSITY_SCALE
			if(cold_fire)
				radiated_temperature = location.air.temperature * COLD_FIRE_SPREAD_RADIOSITY_SCALE
```

File: code/modules/atmospherics/environmental/LINDA_fire.dm [hunk 6]
```diff
@@ -165,7 +176,12 @@
 	var/heat_a = 255
 	var/greyscale_fire = 1 //This determines how greyscaled the fire is.
 
-	if(temperature < 5000) //This is where fire is very orange, we turn it into the normal fire texture here.
+	if(cold_fire)
+		heat_r = 0
+		heat_g = LERP(255, temperature, 1.2)
+		heat_b = LERP(255, temperature, 0.9)
+		heat_a = 100
+	if(temperature < 5000 && temperature > FIRE_MINIMUM_TEMPERATURE_TO_EXIST) //This is where fire is very orange, we turn it into the normal fire texture here.
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_fire.dm:184]:
```suggestion
	else if(temperature < 5000) //This is where fire is very orange, we turn it into the normal fire texture here.
```

File: code/modules/atmospherics/environmental/LINDA_fire.dm [hunk 7]
```diff
@@ -144,6 +153,8 @@
 		temperature = reference.temperature
 
 	// Handles the burning of atoms.
+	if(temperature < FIRE_MINIMUM_TEMPERATURE_TO_EXIST)
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_fire.dm:156]:
```suggestion
	if(cold_fire?)
```

Comment (LemonInTheDark) [reply code/modules/atmospherics/environmental/LINDA_fire.dm:156]: Why not? Order of setting?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/atmospherics/environmental/LINDA_fire.dm (hunk 3 above)
Comment (Ghilker) [reply code/modules/atmospherics/environmental/LINDA_fire.dm:267]: this checks the location.air i don't know if that temperature is equal to the hotspot temperature, so i have no idea if is a good idea to swap them

File: code/modules/atmospherics/environmental/LINDA_fire.dm (hunk 4 above)
Comment (Ghilker) [reply code/modules/atmospherics/environmental/LINDA_fire.dm:300]: missed this

File: code/modules/atmospherics/environmental/LINDA_fire.dm (hunk 7 above)
Comment (Ghilker) [reply code/modules/atmospherics/environmental/LINDA_fire.dm:156]: no you're right i'll change this and the others too

**Pull request merged**